cats-effect: Unable to format file due to bug in scalafmt

This template is a guideline, not a strict requirement.
- **Version**: 2.4.1
- **Integration**: sbt-scalafmt
- **Configuration**:
```
version = 2.4.1
align = none
align.openParenCallSite = true
align.openParenDefnSite = true
maxColumn = 120
continuationIndent.defnSite = 2
assumeStandardLibraryStripMargin = true
danglingParentheses = true
rewrite.rules = [AvoidInfix, SortImports, RedundantBraces, RedundantParens, SortModifiers]
docstrings = JavaDoc
lineEndings=preserve
```

## Steps

Given code like [this](https://github.com/scala-steward/cats-effect/blob/521cf8deaac3cbedb8a6238a8d244a416f8f23f1/laws/shared/src/test/scala/cats/effect/ResourceTests.scala):
```scala
  testAsync("Resource.make is equivalent to a partially applied bracket") { implicit ec =>
    check { (acquire: IO[String], release: String => IO[Unit], f: String => IO[String]) =>
      acquire.bracket(f)(release) <-> Resource.make(acquire)(release).use(f)
    }
  }
```

When I run scalafmt like this:

```bash
sbt lawsJVM/scalafmtAll
```

## Problem

Scalafmt fails:
```
[error] /home/ross/src/cats-effect/laws/shared/src/test/scala/cats/effect/ResourceTests.scala:56: error: Unable to format file due to bug in scalafmt
[error]       acquire.bracket(f)(release) <-> Resource.make(acquire)(release).use(f)
[error]                                 ^: /home/ross/src/cats-effect/laws/shared/src/test/scala/cats/effect/ResourceTests.scala
[error] (lawsJVM / Test / scalafmt) /home/ross/src/cats-effect/laws/shared/src/test/scala/cats/effect/ResourceTests.scala:56: error: Unable to format file due to bug in scalafmt
[error]       acquire.bracket(f)(release) <-> Resource.make(acquire)(release).use(f)
[error]                                 ^: /home/ross/src/cats-effect/laws/shared/src/test/scala/cats/effect/ResourceTests.scala
```

## Expectation

scalafmt succeeds and the output still compiles

## Workaround

Not upgrading to 2.4.1.

## Notes

Similar to #1599, but not the same as what @larsrh reported for cats-effect on 2.4.0.

--- a/scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala
+++ b/scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala
@@ -191,7 +191,18 @@ class Router(formatOps: FormatOps) {
             .get(hash(right))
             .collect {
               case owner: Term.Function =>
-                val arrow = lastLambda(owner).tokens.find(_.is[T.RightArrow])
+                val lastFunc = lastLambda(owner)
+                // look for arrow before body, if any, else after params
+                val arrow = lastFunc.body.tokens.headOption
+                  .map(x => prevNonComment(tokens(x, -1)).left)
+                  .orElse {
+                    val lastParam = lastFunc.params.lastOption
+                    lastParam.flatMap(_.tokens.lastOption).map { x =>
+                      val beforeArrow = nextNonComment(tokens(x))
+                      if (beforeArrow.right.is[T.RightArrow]) beforeArrow.right
+                      else nextNonComment(tokens(beforeArrow, 1)).right
+                    }
+                  }
                 val expire = arrow.getOrElse(owner.params.last.tokens.last)
                 (expire, arrow, 0)
             }

--- a/scalafmt-tests/src/test/resources/default/Lambda.stat
+++ b/scalafmt-tests/src/test/resources/default/Lambda.stat
@@ -301,3 +301,88 @@ object a {
       |""".stripMargin
   }
 }
+<<< #1714 1
+testAsync("Resource.make is equivalent to a partially applied bracket") { implicit ec =>
+    check { (acquire: IO[String], release: String => IO[Unit], f: String => IO[String]) =>
+      acquire.bracket(f)(release) <-> Resource.make(acquire)(release).use(f)
+    }
+  }
+>>>
+testAsync("Resource.make is equivalent to a partially applied bracket") {
+  implicit ec =>
+    check {
+      (acquire: IO[String],
+       release: String => IO[Unit],
+       f: String => IO[String]) =>
+        acquire.bracket(f)(release) <-> Resource.make(acquire)(release).use(f)
+    }
+}
+<<< #1714 2
+align = none
+maxColumn = 120
+danglingParentheses = true
+===
+testAsync("Resource.make is equivalent to a partially applied bracket") { implicit ec =>
+    check { (acquire: IO[String], release: String => IO[Unit], f: String => IO[String]) =>
+      acquire.bracket(f)(release) <-> Resource.make(acquire)(release).use(f)
+    }
+  }
+>>>
+testAsync("Resource.make is equivalent to a partially applied bracket") { implicit ec =>
+  check { (acquire: IO[String], release: String => IO[Unit], f: String => IO[String]) =>
+    acquire.bracket(f)(release) <-> Resource.make(acquire)(release).use(f)
+  }
+}
+<<< #1714 3
+align = none
+maxColumn = 120
+danglingParentheses = true
+===
+testAsync("Resource.make is equivalent to a partially applied bracket") { implicit ec =>
+    check { (acquire: IO[String], release: String => IO[Unit], f: String => IO[String]/* c1 */)/* c2 */=>
+      /*c3*/ acquire.bracket(f)(release) <-> Resource.make(acquire)(release).use(f)
+    }
+  }
+>>>
+testAsync("Resource.make is equivalent to a partially applied bracket") { implicit ec =>
+  check { (acquire: IO[String], release: String => IO[Unit], f: String => IO[String] /* c1 */ ) /* c2 */ =>
+    /*c3*/
+    acquire.bracket(f)(release) <-> Resource.make(acquire)(release).use(f)
+  }
+}
+<<< #1714 4
+align = none
+maxColumn = 120
+danglingParentheses = true
+===
+testAsync("Resource.make is equivalent to a partially applied bracket") { implicit ec =>
+    check { (acquire: IO[String], release: String => IO[Unit], f: String => IO[String]/* c1 */)/* c2 */=>
+      /*c3*/ { acquire.bracket(f)(release) <-> Resource.make(acquire)(release).use(f) }
+    }
+  }
+>>>
+testAsync("Resource.make is equivalent to a partially applied bracket") { implicit ec =>
+  check { (acquire: IO[String], release: String => IO[Unit], f: String => IO[String] /* c1 */ ) /* c2 */ =>
+    /*c3*/
+    { acquire.bracket(f)(release) <-> Resource.make(acquire)(release).use(f) }
+  }
+}
+<<< #1714 5
+align = none
+maxColumn = 120
+danglingParentheses = true
+===
+testAsync("Resource.make is equivalent to a partially applied bracket") { implicit ec =>
+    check { (acquire: IO[String], release: String => IO[Unit], f: String => IO[String]/* c1 */)/* c2 */=>
+      { /*c3*/ acquire.bracket(f)(release) <-> Resource.make(acquire)(release).use(f) }
+    }
+  }
+>>>
+testAsync("Resource.make is equivalent to a partially applied bracket") { implicit ec =>
+  check { (acquire: IO[String], release: String => IO[Unit], f: String => IO[String] /* c1 */ ) /* c2 */ =>
+    {
+      /*c3*/
+      acquire.bracket(f)(release) <-> Resource.make(acquire)(release).use(f)
+    }
+  }
+}

